File: src/main/java/com/gmail/borlandlp/minigamesdtools/creator/RulesProvider.java
package com.gmail.borlandlp.minigamesdtools.creator;

import java.util.*;

public class RulesProvider {
    private List<String> rules = new ArrayList<>();

    public void addReqiredField(String key) {
        this.rules.add(key);
    }

    public void remove(String key) {
        this.rules.remove(key);
    }

    public boolean isCorrectProvider(AbstractDataProvider dataProvider) {
        for (String rulesKey : this.rules) {
            if(!dataProvider.contains(rulesKey)) {
                return false;
            }
        }

        return true;
    }
}

File: src/main/kotlin/com/gmail/borlandlp/minigamesdtools/creator/RulesProvider.kt
package com.gmail.borlandlp.minigamesdtools.creator

import java.util.*

class RulesProvider {
    private val rules: MutableList<String> = ArrayList()

    fun addReqiredField(key: String) {
        rules.add(key)
    }

    fun remove(key: String) {
        rules.remove(key)
    }

    fun isCorrectProvider(dataProvider: AbstractDataProvider): Boolean {
        for (rulesKey in rules) {
            if (!dataProvider.contains(rulesKey)) {
                return false
            }
        }
        return true
    }
}